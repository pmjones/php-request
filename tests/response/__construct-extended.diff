001+ object(ExtResponse)#3 (6) {
002+   ["version":protected]=>
003+   string(3) "1.1"
004+   ["status":protected]=>
005+   int(200)
006+   ["headers":protected]=>
007+   array(0) {
008+   }
009+   ["cookies":protected]=>
010+   array(0) {
011+   }
012+   ["content":protected]=>
001- object(ExtResponse)#1 (6) {
002-   ["version":"SapiResponse":private]=>
004-   ["code":"SapiResponse":private]=>
006-   ["headers":"SapiResponse":private]=>
007-   NULL
008-   ["cookies":"SapiResponse":private]=>
009-   NULL
010-   ["content":"SapiResponse":private]=>
011-   NULL
012-   ["callbacks":"SapiResponse":private]=>
013-   NULL
014+   ["headerCallback":protected]=>
016+ }
017+ array(0) {
015- NULL