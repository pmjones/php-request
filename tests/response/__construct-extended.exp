object(ExtResponse)#1 (6) {
  ["version":"SapiResponse":private]=>
  NULL
  ["code":"SapiResponse":private]=>
  NULL
  ["headers":"SapiResponse":private]=>
  NULL
  ["cookies":"SapiResponse":private]=>
  NULL
  ["content":"SapiResponse":private]=>
  NULL
  ["callbacks":"SapiResponse":private]=>
  NULL
}
NULL