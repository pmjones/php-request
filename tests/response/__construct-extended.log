
---- EXPECTED OUTPUT
object(ExtResponse)#1 (6) {
  ["version":"SapiResponse":private]=>
  NULL
  ["code":"SapiResponse":private]=>
  NULL
  ["headers":"SapiResponse":private]=>
  NULL
  ["cookies":"SapiResponse":private]=>
  NULL
  ["content":"SapiResponse":private]=>
  NULL
  ["callbacks":"SapiResponse":private]=>
  NULL
}
NULL
---- ACTUAL OUTPUT
object(ExtResponse)#3 (6) {
  ["version":protected]=>
  string(3) "1.1"
  ["status":protected]=>
  int(200)
  ["headers":protected]=>
  array(0) {
  }
  ["cookies":protected]=>
  array(0) {
  }
  ["content":protected]=>
  NULL
  ["headerCallback":protected]=>
  NULL
}
array(0) {
}
---- FAILED
