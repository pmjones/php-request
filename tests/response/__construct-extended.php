<?php
class ExtResponse extends SapiResponse
{
    public function __construct()
    {
    }
}
$ext = new ExtResponse();
var_dump($ext);
var_dump($ext->getHeaders());
