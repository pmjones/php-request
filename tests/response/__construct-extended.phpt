--TEST--
SapiResponse::__construct (extended without calling parent)
--FILE--
<?php
class ExtResponse extends SapiResponse
{
    public function __construct()
    {
    }
}
$ext = new ExtResponse();
var_dump($ext);
var_dump($ext->getHeaders());
--EXPECT--
object(ExtResponse)#3 (6) {
  ["version":"SapiResponse":private]=>
  NULL
  ["code":"SapiResponse":private]=>
  NULL
  ["headers":"SapiResponse":private]=>
  NULL
  ["cookies":"SapiResponse":private]=>
  NULL
  ["content":"SapiResponse":private]=>
  NULL
  ["callbacks":"SapiResponse":private]=>
  NULL
}
NULL
