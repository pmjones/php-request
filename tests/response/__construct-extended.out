object(ExtResponse)#3 (6) {
  ["version":protected]=>
  string(3) "1.1"
  ["status":protected]=>
  int(200)
  ["headers":protected]=>
  array(0) {
  }
  ["cookies":protected]=>
  array(0) {
  }
  ["content":protected]=>
  NULL
  ["headerCallback":protected]=>
  NULL
}
array(0) {
}